src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -0,0 +1,293 @@
+            out int tokenLengthOut,
+            out int tokenTypeOut,
+            out int tokenModifiersOut)
+        {
+            // Each semantic token is represented in LSP by five numbers:
+            //     1. Token line number delta, relative to the previous token
+            //     2. Token start character delta, relative to the previous token
+            //     3. Token length
+            //     4. Token type (index) - looked up in SemanticTokensLegend.tokenTypes
+            //     5. Token modifiers - each set bit will be looked up in SemanticTokensLegend.tokenModifiers
+
+            var textSpan = textSpanToClassifiedSpans.Key;
+            var linePosition = lines.GetLinePositionSpan(textSpan).Start;
+            var lineNumber = linePosition.Line;
+            var startCharacter = linePosition.Character;
+
+            // 1. Token line number delta, relative to the previous token
+            var deltaLine = lineNumber - lastLineNumber;
+            Contract.ThrowIfTrue(deltaLine < 0);
+
+            // 2. Token start character delta, relative to the previous token
+            // (Relative to 0 or the previous token’s start if they're on the same line)
+            var startCharacterDelta = startCharacter;
+            if (lastLineNumber == lineNumber)
+            {
+                startCharacterDelta = startCharacter - lastStartCharacter;
+            }
+
+            // 3. Token length
+            var tokenLength = textSpan.Length;
+
+            // Getting the classified spans that are modifiers.
+            // Since a TextSpan can have multiple ClassifiedSpans, all of the ClassifiedSpans except one (the token)
+            // should be a modifier.
+            var modifiers = textSpanToClassifiedSpans.Where(
+                s => ClassificationTypeNames.AdditiveTypeNames.Contains(s.ClassificationType));
+
+            // Filtering out the modifiers for now (will be added in step 5). We just want the primary token.
+            var tokenTypeClassifiedSpan = textSpanToClassifiedSpans.Except(modifiers).Single();
+
+            // 4. Token type - looked up in SemanticTokensLegend.tokenTypes (language server defined mapping
+            // from integer to LSP token types).
+            if (!s_classificationTypeToSemanticTokenTypeMap.TryGetValue(tokenTypeClassifiedSpan.ClassificationType, out var tokenTypeStr))
+            {
+                throw new NotSupportedException($"Classification type {tokenTypeClassifiedSpan.ClassificationType} is unsupported.");
+            }
+
+            var tokenTypeIndex = GetTokenTypeIndex(tokenTypeStr);
+
+            // 5. Token modifiers - each set bit will be looked up in SemanticTokensLegend.tokenModifiers
+            var modifierBits = TokenModifiers.None;
+            foreach (var currentModifier in modifiers)
+            {
+                if (!s_classificationTypeToSemanticTokenModifierMap.TryGetValue(currentModifier.ClassificationType, out var modifier))
```

Comment:
1. this seems like massive overkill given we have one modifier.
2. we'll throw an exception if anotehr modifier is ever added and we don't test lsp.  this feels not good.

Reply:
Would you suggest we just explicitly test for the 'static' modifier and remove all other logic? My only concern is that if another modifier is ever added, it will probably be very easy to miss updating LSP.
Another option could be to change the exception to a Debug assert?

Reply:
yup, i'd check for the single case, and assert if we hit something else.  wherever the 'static' guy is defined, we can also put a comment saying that LSp woul dneed to be updated here.